dask/array/utils.py

```diff
@@ -388,6 +388,40 @@ def zeros_like_safe(a, shape, **kwargs):
         return np.zeros(shape, **kwargs)
 
 
+def _array_like_safe(np_func, da_func, a, like, **kwargs):
+    if like is a and hasattr(a, "__array_function__"):
+        return a
+
+    if isinstance(like, Array):
+        return da_func(a, **kwargs)
+    elif isinstance(a, Array):
+        try:
+            import cupy as cp
```

Comment:
Would move this `import` to the global level using something like this...

```python
try:
    import cupy
except ImportError:
    cupy = None
```

Then we can just check to see `if cupy is not None:`

Reply:
There was another place with a similar use, I attempted to unify both cases in https://github.com/dask/dask/pull/7162/commits/eb6e0fa2ef4b4dcd479c68fe364f83a16570b9f2. If you see any issues with that please let me know and I'll change that again.

Reply:
Nope that looks great. Thanks Peter! 😄